services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java

```diff
@@ -210,7 +211,7 @@ void initBuiltIns() {
                 true, true);
         builtins.put(UPN, model);
 
-        model = UserRealmRoleMappingMapper.create(null, GROUPS, GROUPS, true, true, true);
```

Comment:
I'm afraid you can't change this one because this built-in mapper is used when you have MicroProfile-based clients. There, the `realm roles` (not group membership) is added to a specific `groups` claim in the token.

See https://github.com/keycloak/keycloak/pull/5933.

Reply:
Yeah, in MP-JWT the roles are added to the `groups` claim.